core/cap-0046-06.md

```diff
@@ -198,20 +205,27 @@ compliance functionality.
 * Admin Interface
 *
 \******************************************************************************/
-// If "admin" is the administrator, burn "amount" from "from"
-fn burn(admin: Signature, nonce: BigInt, from: Identifier, amount: BigInt) -> Result<(), Error>;
+// If "admin" is the administrator and clawback is not disabled, clawback "amount" from "from".
+// The "amount" is burned. This function can be used on deauthorized balances. 
+fn clawback(admin: Signature, nonce: BigInt, from: Identifier, amount: u128) -> Result<(), Error>;
```

Comment:
I don't have a strong opinion, but I wonder what is the reason for moving from terms that exist in smart token world to the classic asset terms. Now we have somewhat inconsistent interface (e.g. we have `mint` and not `issue` and the contract itself is `token` and not `asset` etc).

Reply:
Some of the Soroban terms make more sense in the classic sense. There are specific clawback operations in classic and a clawback method in Soroban. There is no issue operation in classic (you either issue through a payment or through an offer), which doesn't work on Soroban, so we have an explicit mint method instead.

Re:token vs asset - We've been trying to differentiate the Stellar classic vs Soroban line by using classic asset vs soroban or smart token. I don't think I've done a good job of this, and it's possible this CAP is out of date when it comes to this terminology.

Reply:
I mean, I'm fine with whatever terminology, just pointed that it seemed like it's less consistent. Though now that I've learnt that burn did something unexpected anyway, I think that's a reasonable change to make.